From 59a009787ca913f541496173bb9fa933c2e9f101 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Wed, 8 Oct 2025 01:03:24 +0300
Subject: [PATCH 342/553] tst_QAccessibility: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: Ib52a996329a07be5accf555d6228a57a8764d87c
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit 9f82f972201184b63b7b5edb9306761bcc1ac49d)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/other/qaccessibility/tst_qaccessibility.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/auto/other/qaccessibility/tst_qaccessibility.cpp b/tests/auto/other/qaccessibility/tst_qaccessibility.cpp
index 64d216e33dd..b0c4516f964 100644
--- a/tests/auto/other/qaccessibility/tst_qaccessibility.cpp
+++ b/tests/auto/other/qaccessibility/tst_qaccessibility.cpp
@@ -2628,10 +2628,10 @@ void tst_QAccessibility::groupBoxTest()
     QCOMPARE(iface->role(), QAccessible::Grouping);
     QCOMPARE(iface->text(QAccessible::Name), QLatin1String("Test QGroupBox"));
     QCOMPARE(iface->text(QAccessible::Description), QLatin1String("This group box will be used to test accessibility"));
-    QList<QPair<QAccessibleInterface *, QAccessible::Relation>> relations =
+    QList<std::pair<QAccessibleInterface *, QAccessible::Relation>> relations =
             rButtonIface->relations();
     QCOMPARE(relations.size(), 1);
-    QPair<QAccessibleInterface*, QAccessible::Relation> relation = relations.first();
+    auto relation = relations.first();
     QCOMPARE(relation.first->object(), groupBox);
     QCOMPARE(relation.second, QAccessible::Label);
     }
@@ -3914,7 +3914,7 @@ void tst_QAccessibility::relationTest()
     QAccessibleInterface *acc_pb = QAccessible::queryAccessibleInterface(pb);
     QVERIFY(acc_pb);
 
-    typedef QPair<QAccessibleInterface*, QAccessible::Relation> RelationPair;
+    typedef std::pair<QAccessibleInterface*, QAccessible::Relation> RelationPair;
     {
         const QList<RelationPair> rels = acc_label->relations(QAccessible::Labelled);
         QCOMPARE(rels.size(), 1);
@@ -3982,7 +3982,7 @@ void tst_QAccessibility::labelTest()
     QCOMPARE(acc_label->state().readOnly, true);
 
 
-    typedef QPair<QAccessibleInterface*, QAccessible::Relation> RelationPair;
+    using RelationPair = std::pair<QAccessibleInterface*, QAccessible::Relation>;
     {
         const QList<RelationPair> rels = acc_label->relations(QAccessible::Labelled);
         QCOMPARE(rels.size(), 1);
-- 
2.51.2

